source/client/benchmark_client_impl.cc

```diff
@@ -68,22 +68,12 @@ StatisticPtrMap BenchmarkClientHttpImpl::statistics() const {
 };
 
 bool BenchmarkClientHttpImpl::tryStartRequest(CompletionCallback caller_completion_callback) {
-  // When we allow client-side queuing, we want to have a sense of time spend waiting on that queue.
-  // So we return false here to indicate we couldn't initiate a new request.
   auto* pool_ptr = pool();
-  auto cluster_info = cluster();
-  if (pool_ptr == nullptr || cluster_info == nullptr ||
-      !cluster_info->resourceManager(Envoy::Upstream::ResourcePriority::Default)
-           .pendingRequests()
-           .canCreate()) {
-    return false;
-  }
-  // When no client side queueing is disabled (max_pending equals 1) we control the pacing as
-  // exactly as possible here.
-  // NOTE: We can't consistently rely on resourceManager()::requests()
-  // because that isn't used for h/1 (it is used in tcp and h2 though).
-  if ((max_pending_requests_ == 1 &&
-       (requests_initiated_ - requests_completed_) >= connection_limit_)) {
+  const uint64_t max_in_flight =
+      max_pending_requests_ + (use_h2_ ? max_active_requests_ : connection_limit_);
+  if ((requests_initiated_ - requests_completed_ >= max_in_flight) || pool_ptr == nullptr) {
```

Comment:
Nit: The pool_ptr == nullptr check might be clearer in a separate `if`, directly after pool_ptr was defined.

Reply:
Also done in 0fcaf9185769ad68e07c51c9e3fee55e0035e6f3